### PR TITLE
Rewrite documentation about building

### LINKED ISSUE #447: Document how build process works

### PR DESCRIPTION
**Pull request by goldmann**
Fixes #447

### CODE REVIEW

File: cekit/descriptor/module.py [hunk 1]
```diff
@@ -24,7 +24,8 @@ def __init__(self, descriptor, path, artifact_dir):
         self.skip_merging = ['description',
                              'version',
                              'name',
-                             'release']
+                             'release',
+                             'help']
```
Comment (goldmann) [inline cekit/descriptor/module.py]: @rnc This PR additionally adds a change that was missed in https://github.com/cekit/cekit/pull/445

**Comment by goldmann** [comment]
Generated document can be found here: https://1061-102005331-gh.circle-artifacts.com/0/home/circleci/project/docs/_build/html/build.html

File: docs/build.rst [hunk 2]
```diff
@@ -1,117 +1,344 @@
...
-**Sub-commands that are available are:**
+Next step is to apply module overrides to the image object we have. Modules are
+actually overrides with the difference that modules encapsulate a defined functionality whereas
+overrides are just modifying things.
 
-* ``buildah`` --  Build using Buildah engine
-* ``docker`` -- Build using Docker engine
-* ``osbs`` -- Build using OSBS engine
-* ``podman`` -- Build using Podman engine
+To do this we iterate over all modules defined to install we try to find them in the module registry
```
Comment (rnc) [inline docs/build.rst]: Should this be : ? 
```suggestion
To do this we iterate over all modules that are defined to install and we try to find them in the module registry
```

File: docs/build.rst [hunk 3]
```diff
@@ -1,117 +1,344 @@
...
+************************
 
-.. code:: bash
+Final step is to execute the build using selected builder.
 
-	  $ cekit build
+Resulting image sometimes will be available on your localhost, sometimes in some remote
+registry. It all depends on the builder.
 
+Common ``build`` parameters
```
Comment (rnc) [inline docs/build.rst]: Should ``build`` be in code style ? - or have it all in heading style ?

File: cekit/descriptor/module.py (hunk 1 above)
Comment (rnc) [reply cekit/descriptor/module.py]: Ok

File: docs/build.rst (hunk 3 above)
Comment (goldmann) [reply docs/build.rst]: I think it should be without code style, I'll change it.

**Pull request merged**